tools/uf2conv.py

```diff
@@ -252,21 +269,12 @@ def get_drives():
             if len(words) >= 3 and words[1] == "2" and words[2] == "FAT":
                 drives.append(words[0])
     else:
-        rootpath = "/run/media"
-        if not os.path.isdir(rootpath):
-            rootpath = "/media"
-        if not os.path.isdir(rootpath):
-            rootpath = "/opt/media"
         if sys.platform == "darwin":
-            rootpath = "/Volumes"
+            possibly_anydir("/Volumes", drives)
         elif sys.platform == "linux":
-            tmp = rootpath + "/" + os.environ["USER"]
-            if os.path.isdir(tmp):
-                rootpath = tmp
-        for d in os.listdir(rootpath):
-            drives.append(os.path.join(rootpath, d))
-
-        if (len(drives) == 0) and (sys.platform == "linux"):
+            # Do first since a race condition is possible.
+            time.sleep(2)
```

Comment:
It still seems like a race condition possibility if you think udev is going to barf if we try and access a dir it's in the process of creating.

Is is possible to put the sleep() at the end of the check, only if it fails?  OTW we're adding another 2 seconds to every upload ever which gets to be annoying....

Reply:
The only case that makes sense to sleep after is if the boot button was pressed while plugging in, and is not the normal or usual flow.

This is actually an optimization for the pico just being reset.
The udev and the USB subsystem combined usually take between 2 to 4 seconds to finish mounting, even on my 16core box. chewing up cpu then waiting doesn't make sense, loop will happen twice. 

Therefore, to satisfy both you would have to pass a variable indicating that the reset failed, or was successful, and then decide if to delay at the beginning or on failure. I did not want to do too many changes and make the patch too large.